AdminWebsite/AdminWebsite/ClientApp/src/app/booking/assign-judge/assign-judge.component.ts

```diff
@@ -87,37 +93,74 @@ export class AssignJudgeComponent extends BookingBaseComponent implements OnInit
     }
 
     private initForm() {
-        const find_judge = this.hearing.participants.find(x => x.is_judge === true);
-        if (!find_judge) {
-            this.judge = new JudgeResponse({ email: this.constants.PleaseSelect, display_name: '' });
-        } else {
-            this.logger.debug(`${this.loggerPrefix} Found judge in hearing. Populating existing selection.`);
-            this.judge = AssignJudgeComponent.mapJudge(find_judge);
-            this.canNavigate = true;
-        }
-        this.judgeDisplayName = new FormControl(this.judge.display_name, {
-            validators: [Validators.required, Validators.pattern(Constants.TextInputPattern), Validators.maxLength(255)],
-            updateOn: 'blur'
-        });
+        const existingJudge = this.hearing.participants.find(x => x.is_judge === true);
+
+        this.populateFormFields(existingJudge);
 
         this.form = this.fb.group({
-            judgeName: [this.judge.email, Validators.required],
-            judgeDisplayName: this.judgeDisplayName
+            judgeName: [this.judge.display_name, Validators.required],
+            judgeDisplayNameFld: this.judgeDisplayNameFld,
+            judgeEmailFld: this.judgeEmailFld,
+            judgePhoneFld: this.judgePhoneFld
         });
 
+        this.updateJudgeSummary();
+    }
+
+    updateJudgeSummary() {
         this.$subscriptions.push(
             this.judgeName.valueChanges.subscribe(judgeUserId => {
                 this.addJudge(judgeUserId);
                 this.isJudgeSelected = judgeUserId !== null;
+                this.judgeEmail = '';
+                this.judgeEmailFld.setValue('');
+                this.judgePhone = '';
+                this.judgePhoneFld.setValue('');
                 this.canNavigate = this.isJudgeSelected;
             })
         );
 
         this.$subscriptions.push(
-            this.judgeDisplayName.valueChanges.subscribe(name => {
+            this.judgeDisplayNameFld.valueChanges.subscribe(name => {
                 this.judge.display_name = name;
             })
         );
+
+        this.$subscriptions.push(
+            this.judgeEmailFld.valueChanges.subscribe(email => {
+                this.judgeEmail = email;
+            })
+        );
+
+        this.$subscriptions.push(
+            this.judgePhoneFld.valueChanges.subscribe(phone => {
+                this.judgePhone = phone;
+            })
+        );
+    }
+
+    populateFormFields(existingJudge: ParticipantModel) {
+        if (!existingJudge) {
+            this.judge = new JudgeResponse({ email: this.constants.PleaseSelect, display_name: '' });
```

Comment:
The new & JudgeResponse are redundant here, just init the object

Reply:
If I do 
`this.judge = { email: this.constants.PleaseSelect, display_name: '' } `
then I get an error saying type is missing init and toJSON

Reply:
Ok keeping this one as it uses the JudgeResponse constructor